cirq-core/cirq/experiments/xeb_fitting.py

```diff
@@ -66,18 +66,37 @@ def benchmark_2q_xeb_fidelities(
          sampled_df: The sampled results to benchmark. This is likely produced by a call to
             `sample_2q_xeb_circuits`.
         circuits: The library of circuits corresponding to the sampled results in `sampled_df`.
-        cycle_depths: The sequence of cycle depths to simulate the circuits.
+        cycle_depths: The sequence of cycle depths to benchmark the circuits. If not provided,
+            we use the cycle depths found in `sampled_df`. All requested `cycle_depths` must be
+            present in `sampled_df`.
         param_resolver: If circuits contain parameters, resolve according to this ParamResolver
             prior to simulation
         pool: If provided, execute the simulations in parallel.
 
     Returns:
         A DataFrame with columns 'cycle_depth' and 'fidelity'.
     """
+    sampled_cycle_depths = (
+        sampled_df.index.get_level_values('cycle_depth').drop_duplicates().sort_values()
+    )
+    if cycle_depths is not None:
+        if len(cycle_depths) == 0:
+            raise ValueError("`cycle_depths` should be a non-empty array_like")
+        not_in_sampled = np.setdiff1d(cycle_depths, sampled_cycle_depths)
+        if len(not_in_sampled) > 0:
+            raise ValueError(
+                f"The `cycle_depths` provided include some not "
+                f"available in `sampled_df`: {not_in_sampled}"
+            )
+        sim_cycle_depths = np.intersect1d(cycle_depths, sampled_cycle_depths)
```

Comment:
Isn't the interesect1d redundant here? sampled_cycle_depths is always a superset of  cycle_depths because of the previous condition, and result is always cycle_depths?

Reply:
yes that's true. removed